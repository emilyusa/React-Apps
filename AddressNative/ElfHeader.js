import React from 'react';
import {Link} from 'react-router-native';
import {Text, View} from 'react-native';
import styles from "./elf-styles";

export default class ElfHeader extends React.Component {

    render() {
        return (
            <View style={styles.nav}>
                <Link
                    to="/"
                    underlayColor='#f0f4f7'
                    style={styles.navItem}>
                    <Text>Address</Text>
                </Link>
                <Link
                    to="/get-file"
                    underlayColor='#f0f4f7'
                    style={styles.navItem}>
                    <Text>GetFile</Text>
                </Link>
            </View>
        );
    }
}

