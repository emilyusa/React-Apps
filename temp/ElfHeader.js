import React from 'react';
import {Link} from 'react-router-native';
import {Text, View} from 'react-native';
import styles from "./elf-styles";


class ElfHeader extends React.Component {
	

    render() {
        return (
            <View style={styles.nav}>
                <Link
                    to="/"
                    underlayColor='#f0f4f7'
                    style={styles.navItem}>
                    <Text>Address</Text>
                </Link>
                /<Link
                    to="/"
                    underlayColor='#f0f4f7'
                    style={styles.navItem}>
                    <Text>GetFile</Text>
                </Link>
            </View>
        );
    }
}

export default ElfHeader;

