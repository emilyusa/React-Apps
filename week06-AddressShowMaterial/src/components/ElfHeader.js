import { Link } from 'react-router-dom';
import React, { Component } from 'react';
import '../css/menu.css';
import AppBar from 'material-ui/AppBar';
import Drawer from 'material-ui/Drawer';
import MenuItem from 'material-ui/MenuItem';

class ElfHeader extends Component {

    constructor(props) {
        super(props);
        this.state = {
            open: false
        };
    }

    handleToggle = () => this.setState({ open: !this.state.open });

    render() {
        return (			
			  <div className="menu">
                  <div>
                      <AppBar
                          title="Address Maven"
                          iconClassNameRight="muidocs-icon-navigation-expand-more"
                          onLeftIconButtonClick={this.handleToggle}
                      />
                      <Drawer
                          docked={false}
                          width={200}
                          open={this.state.open}
                          onRequestChange={this.handleToggle}
                      >
                          <AppBar title="Address Maven"/>

                          <MenuItem
                              primaryText='Address'
                              containerElement={<Link to="/"/>}
                              onClick={this.handleToggle}
                          />

                          <MenuItem
                              primaryText='Get File'
                              containerElement={<Link to="/get-file"/>}
                              onClick={this.handleToggle}
                          />
                      </Drawer>
                  </div>
			  </div>                
			
			
        );
    }
}


export default ElfHeader;
