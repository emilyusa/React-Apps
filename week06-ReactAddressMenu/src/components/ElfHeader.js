import { Link } from 'react-router-dom';
import React, { Component } from 'react';
import Header from './Header';

class ElfHeader extends Component {
    render() {
        return (			
			  <div className="App">
			      <Header />
				  <ul>
					  <li><Link to="/">Address</Link></li>
					  <li><Link to="/get-file">Get File</Link></li>          
				  </ul>
				  
			  </div>                
			
			
        );
    }
}


export default ElfHeader;
