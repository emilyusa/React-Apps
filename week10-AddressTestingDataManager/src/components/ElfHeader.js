import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Drawer from '@material-ui/core/Drawer';
import IconButton from '@material-ui/core/IconButton';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import MenuIcon from '@material-ui/icons/Menu';
import Divider from '@material-ui/core/Divider';
import List from '@material-ui/core/List';
import { gitItems, demoItems, initialDB } from './tileData';


const styles = {
    root: {
        flexGrow: 1
    },
    flex: {
        flex: 1
    },
};


class ElfHeader extends Component {
    constructor(props) {
        super(props);
        this.state = {
            open: false
        };
    }

    handleToggle = () => this.setState({ open: !this.state.open });

    render() {
        const { classes } = this.props;
        const sideList = (
            <div className={classes.list}>
                <List>{gitItems}</List>
                <Divider />
                <List>{demoItems}</List>
                <Divider/>
                <List>{initialDB}</List>
            </div>
        );
        return (
            <div>
                <AppBar
                    position="static"
                >
                    <Toolbar>
                        <IconButton
                            className={classes.menuButton}
                            color="inherit"
                            aria-label="Menu"
                            onClick={this.handleToggle}
                        >
                            <MenuIcon/>
                        </IconButton>
                        <Typography variant="title" color="inherit" className={classes.flex}>
                            Address Maven
                        </Typography>
                    </Toolbar>
                </AppBar>
                <Drawer open={this.state.open} onClose={this.handleToggle}>
                    <div
                        role="button"
                        onClick={this.handleToggle}
                        onKeyDown={this.handleToggle}
                    >
                        {sideList}
                    </div>
                </Drawer>

            </div>
        );
    }
}

ElfHeader.propTypes = {
    classes: PropTypes.object.isRequired
};

export default withStyles(styles)(ElfHeader);