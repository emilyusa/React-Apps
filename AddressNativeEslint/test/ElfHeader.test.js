import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import ElfHeader from '../ElfHeader';



configure({adapter:new Adapter()});

describe('jest test ElfHeader', function() {

    it('renders without crashing', () => {
        const tree = renderer.create(<ElfHeader />).toJSON();
        expect(tree).toMatchSnapshot();

    });


});