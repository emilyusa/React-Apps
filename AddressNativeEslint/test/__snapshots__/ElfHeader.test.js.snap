// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest test ElfHeader renders without crashing 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-around",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "padding": 10,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={null}
    >
      Address
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "padding": 10,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={null}
    >
      GetFile
    </Text>
  </View>
</View>
`;
